tsl/src/remote/tuplefactory.c

```diff
@@ -328,7 +327,10 @@ tuplefactory_make_virtual_tuple(TupleFactory *tf, PGresult *res, int row, int fo
 				if (format == FORMAT_TEXT)
 					datum = DirectFunctionCall1(tidin, CStringGetDatum(valstr));
 				else
-					datum = DirectFunctionCall1(tidrecv, PointerGetDatum(buf));
+				{
+					StringInfoData si = {.data = valstr, .len = len};
+					datum = DirectFunctionCall1(tidrecv, PointerGetDatum(&si));
+				}
```

Comment:
Hmmm... this is not really what is recommended in PG but I see why you want to do it and is using `StringInfoData` as a [`std::string_view`](https://en.cppreference.com/w/cpp/string/basic_string_view). Since the alternative would be to build up a complete `StringInfo` buffer by copying `valstr`, I think this would be OK. This requires that the lifetime of `valstr` is longer than the function call, which it seems to be.